lib/tokenizer/index.js

```diff
@@ -2135,10 +2135,17 @@ _[NAMED_CHARACTER_REFERENCE_STATE] = function (cp) {
     else {
         if (matchResult) {
             var lastConsumed = this.tempBuff[this.tempBuff.length - 1];
+            var isSemicolon = lastConsumed === $.SEMICOLON;
+            var quirkAttr = this._isCharacterReferenceAttributeQuirk();
 
-            if (lastConsumed === $.SEMICOLON || !this._isCharacterReferenceAttributeQuirk())
+            if (!isSemicolon && !quirkAttr)
+                this._err(ERR.missingSemicolonAfterCharacterReference);
```

Comment:
I'm looking at tests and we have not quite correct error location for this error. It refers to the last symbol of the entity, e.g.: in `I'm &notin` it points to `t`, however it makes more sense if it will point to `i` since it's the place there semicolon is expected. I sugges to introduce new tokenizer method:
```js
Tokenizer.prototype._errOnNextCodepoint = function(code) {
    this._consume();
    this._err(code);
    this._unconsume();
};
```

and use it here. We need to update tests as well.